### PR TITLE
(PDS-584) Overlay component

### PR DESCRIPTION
**Pull request by mardotio**
### Overview

Adds `Overlay` component that can place any content on top of the existing view. The component is built using [PopperJS](https://popper.js.org/docs/v2/).

The overlay is positioned relative to a target element, and the final position can be adjusted using the `align` and `position` properties. The position of the overlay can also be overwritten by passing and offset function or tuple (https://popper.js.org/docs/v2/modifiers/offset/). By default, the overlay will be rendered within the same DOM element where it is defined, but if a container ref is provided, the overlay can be injected into any DOM element.

#### Positioning

https://user-images.githubusercontent.com/25378324/113339062-f2062b00-92f7-11eb-9d1b-6b62710bbf18.mov

#### Confirm delete

https://user-images.githubusercontent.com/25378324/113339056-ee72a400-92f7-11eb-90bb-8bfa91f63d74.mov

### CODE REVIEW

**Review by chrisleicester — APPROVED** [review]
This is really awesome work. Thanks for tackling this.

I think its complex enough that it deserves another set of eyes before merging, but totally up to you.

**Review by jilliankeenan — APPROVED** [review]

**Pull request merged**
